src/Orleans.Transactions/State/TransactionalState.cs

```diff
@@ -501,30 +482,59 @@ public override string ToString()
         {
             return $"{this.context.GrainInstance}.{this.config.StateName}";
         }
-    }
 
-    [Serializable]
-    public class LogRecord<T>
-    {
-        public T NewVal { get; set; }
-        public TransactionalResourceVersion Version { get; set; }
-    }
+        [Serializable]
+        private class Metadata
```

Comment:
Do you need to mark this class Serializable? I don't think this is relying on binary formatter nor Orleans serializers, right? or is there a path where those might be used?

Reply:
We've gone back and forth a couple of times.  It's not needed in this incarnation, but it does no harm either.